src/test/java/org/cactoos/map/MapEnvelopeTest.java

```diff
@@ -46,93 +43,73 @@
-            new NoNulls<>(
+            "put method must throw exception",
+            () -> new NoNulls<>(
                 new MapOf<Integer, Integer>(
-                    new MapEntry<Integer, Integer>(0, -1)
+                    new MapEntry<>(0, -1)
                 )
-            ),
-            new MatcherOf<>(
-                new FuncOf<>(
-                    (map) -> map.put(2, 2),
-                    true
-                ))
+            ).put(2, 2),
+            new Throws<>(
+                "#put() is not supported, it's a read-only map",
+                UnsupportedOperationException.class
+            )
         );
     }
 
     @Test
     public void removeThrowsException() {
-        this.exception.expect(UnsupportedOperationException.class);
-        this.exception.expectMessage(
-            "#remove() is not supported, it's a read-only map"
-        );
         MatcherAssert.assertThat(
             "remove method did not throw exception",
-            new NoNulls<>(
+            () -> new NoNulls<>(
                 new MapOf<Integer, Integer>(
                     new MapEntry<>(0, -1)
                 )
-            ),
-            new MatcherOf<>(
-                new FuncOf<>(
-                    (map) -> map.remove(0),
-                    true
-                ))
+            ).remove(0),
+            new Throws<>(
+                "#remove() is not supported, it's a read-only map",
+                UnsupportedOperationException.class
+            )
         );
     }
 
     @Test
     public void putAllThrowsException() {
-        this.exception.expect(UnsupportedOperationException.class);
-        this.exception.expectMessage(
-            "#putAll() is not supported, it's a read-only map"
-        );
         MatcherAssert.assertThat(
-            "putAll method did not throw exception",
-            new NoNulls<>(
-                new MapOf<Integer, Integer>(
-                    new MapEntry<>(0, -1)
-                )
-            ),
-            new MatcherOf<>(
-                new FuncOf<>(
-                    (map) -> map.putAll(new MapOf<Integer, Integer>()),
-                    true
-                ))
+            "putAll method must throw exception",
+            () -> {
+                new NoNulls<>(
+                    new MapOf<Integer, Integer>(
+                        new MapEntry<>(0, -1)
+                    )
+                ).putAll(new MapOf<Integer, Integer>());
+                return 0;
+            },
+            new Throws<>(
+                "#putAll() is not supported, it's a read-only map",
+                UnsupportedOperationException.class
+            )
         );
     }
 
     @Test
     public void clearThrowsException() {
-        this.exception.expect(UnsupportedOperationException.class);
-        this.exception.expectMessage(
-            "#clear() is not supported, it's a read-only map"
-        );
         MatcherAssert.assertThat(
```

Comment:
@atapin please use `Assertion` instead of the static methods from `MatcherAssert`